client/pages/clubs/[id].tsx

```diff
@@ -1,22 +1,275 @@
+
+interface BannerProps {
+  backgroundImageUrl: string
+}
+
+const largerThan = (size: string): string => `(min-width: ${size})`
+
+const Banner = styled.div<BannerProps>`
+  background-image: url(${({ backgroundImageUrl }) => backgroundImageUrl || ''});
+  background-position: center;
+  height: 20vh;
+
+  @media ${largerThan(tablet)} {
+    height: 30vh;
+  }
+`
+
+const Container = styled.div`
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  margin: 0 auto 100px;
+
+  width: 85%;
+
+  @media ${largerThan(mobile)} {
+    width: 65%;
+  }
+`
 
 export default function ClubDetail() {
-  const { clubs } = useAppContext();
-  const router = useRouter();
+  const { clubs } = useAppContext()
+  const router = useRouter()
 
-  const { id } = router.query;
-  if (typeof id !== 'string') return null;
+  const { id } = router.query
+  if (typeof id !== 'string') return null
 
-  const { name, description } = clubs.get(parseInt(id));
+  const club = clubs.get(parseInt(id))
 
   return (
     <div>
-      <h1>Club Details</h1>
-      <p>{id}</p>
-      <p>{name}</p>
-      <p>{description}</p>
+      <Banner backgroundImageUrl={club.backgroundImageURL} />
+      <Container>
+        <ClubInfo club={club} />
+        <EventsHostedByClub events={club.events} />
+      </Container>
     </div>
-  );
+  )
+}
+
+const BackArrow = styled.img`
+  margin-top: 64px;
+  cursor: pointer;
+`
+
+const Logo = styled.img`
+  border-radius: 100%;
+  display: block;
+  margin-top: 40px;
+  height: 100px;
+  width: 100px;
+`
+
+const ClubName = styled(PageTitle)`
+  margin-bottom: 0;
+  margin-top: 32px;
+`
+
+const Links = styled.div`
+  display: flex;
+  margin-top: 16px;
+`
+
+const SocialLink = styled.a`
+  text-decoration: underline;
+`
+
+const ClubMetaData = styled.div`
+  align-items: center;
+  display: flex;
+  justify-content: space-between;
+  margin-top: 16px;
+`
+
+const MemberCount = styled.div`
+  font-style: italic;
+  font-weight: ${fontWeight.bold};
+`
+
+const CategoriesWrapper = styled.div`
+  display: flex;
+  margin-right: -4px;
+`
+
+const Category = styled.p`
+  border: 1px solid black;
+  border-radius: 15px;
+  margin: 0 4px;
+
+  padding: 4px 10px;
+  font-size: 14px;
+
+  @media ${largerThan(mobile)} {
+    font-size: 16px;
+  }
+`
+
+const ClubDescription = styled.p`
+  line-height: 1.3;
+  margin-top: 24px;
+  margin-bottom: 24px;
+`
+
+interface ClubInfoProps {
+  club: Club
 }
+
+const ClubInfo = ({ club }: ClubInfoProps) => {
+  const router = useRouter()
+
+  const { name, description, iconURL, tags } = club
+  const links = [club.facebookLink, club.twitterLink, club.instagramLink, club.websiteLink]
+  const linksThatExist = links.filter((l) => l !== undefined)
+
+  return (
+    <div>
+      <BackArrow src="/back-arrow.svg" width="28px" onClick={() => router.back()} />
+
+      <Logo src={iconURL} alt="" />
+      <ClubName> {name}</ClubName>
+
+      <Links>
+        {linksThatExist.map((l) => (
```

Comment:
might get JS warnings about rendered elements not containing keys here. Might be worthwhile to suppress or our console will get really red. I followed the convention of including dashed component name as prefix to any unique identifier: 
![image](https://user-images.githubusercontent.com/21977243/116594504-4816c000-a8f0-11eb-8ddf-a3148cdc85d8.png)

Reply:
great catch